wfl/Database/MongoDBManager.cs

```diff
@@ -19,62 +20,66 @@ public MongoDBManager()
             db = client.GetDatabase("WFL");
         }
 
-        public async Task<Restaurant> InsertRestaurantAsync(RestaurantDto r)
+        private string GenerateCollectionName<T>() where T: new()
         {
-            IMongoCollection<Restaurant> collection = db.GetCollection<Restaurant>("restaurants");
-            Restaurant restaurant = new Restaurant()
+            T t = new T();
+            string type = t.GetType().ToString();
+            string collectionName = "";
+            bool started = true;
+            foreach(char c in type)
             {
-                Name = r.Name,
-                MenuLink = r.MenuLink
-            };
-            await collection.InsertOneAsync(restaurant);
-            return restaurant;
-        }
+                if(c == '.')
+                {
+                    collectionName = "";
+                    started = false;
+                }
+                else if (char.IsUpper(c))
+                {
+                    if (started == true)
+                    {
+                        collectionName += '-';
+                    }
+                    else
+                        started = true;
+                    collectionName += char.ToLower(c);
+                }
+                else
+                    collectionName += c;
+            }
+            collectionName += 's';
+            Console.WriteLine(collectionName);
+            return collectionName;
 
-        public async Task<Restaurant> UpdateRestaurantAsync(string id, RestaurantDto r)
-        {
-            IMongoCollection<Restaurant> collection = db.GetCollection<Restaurant>("restaurants");
-            var filter = Builders<Restaurant>.Filter.Eq("Id", id);
-            Restaurant restaurant = new Restaurant
-            {
-                Name = r.Name,
-                MenuLink = r.MenuLink
-            };
-            await collection.ReplaceOneAsync(filter, restaurant);
-            IAsyncCursor<Restaurant> task = await collection.FindAsync(filter);
-            return await task.FirstAsync();
         }
 
-        public async Task<IEnumerable<Restaurant> > GetAllRestaurantsAsync()
+        public async Task<T> InsertEntityAsync<T>(T entity) where T: IEntity, new()
```

Comment:
I'm constructing an object of the generic type when calling `GenerateCollectionName`.